public/app/features/inspector/InspectJSONTab.tsx

```diff
@@ -217,5 +217,18 @@ export class InspectJSONTab extends PureComponent<Props, State> {
 }
 
 function getPrettyJSON(obj: any): string {
-  return JSON.stringify(obj, null, 2);
+  let r = '';
+  try {
+    r = JSON.stringify(obj, null, 2);
+  } catch (e) {
+    if (
+      e instanceof Error &&
+      (e.toString().includes('RangeError') || e.toString().includes('allocation size overflow'))
+    ) {
+      appEvents.emit(AppEvents.alertError, [e.toString(), 'possibly too much data, try setting a lower line limit']);
```

Comment:
Since this code (`getPrettyJSON`) is used not only in Explore's inspect, I don't think we should provide the error message that indicates a resolution to the problem. Can you make it a little bit more generic, I.e. "Cannot display JSON, the object is too big."